public/app/features/alerting/unified/state/actions.ts

```diff
@@ -143,89 +113,18 @@ export function fetchPromAndRulerRulesAction({
   matcher?: Matcher[];
   state?: string[];
 }): ThunkResult<Promise<void>> {
-  return async (dispatch, getState) => {
-    await dispatch(fetchRulesSourceBuildInfoAction({ rulesSourceName }));
-    const dsConfig = getDataSourceConfig(getState, rulesSourceName);
+  return async (dispatch) => {
+    const { data: dsFeatures } = await dispatch(
+      featureDiscoveryApi.endpoints.discoverDsFeatures.initiate({ rulesSourceName })
+    );
 
     await dispatch(fetchPromRulesAction({ rulesSourceName, identifier, filter, limitAlerts, matcher, state }));
-    if (dsConfig.rulerConfig) {
+    if (dsFeatures?.rulerConfig) {
       await dispatch(fetchRulerRulesAction({ rulesSourceName }));
```

Comment:
nit: could we send both of these requests at the same time? So we don't wait for the `promRules` to complete before fetching `rulerRules`?